src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java

```diff
@@ -172,10 +174,27 @@ public static GenericStatus computeChunkStatus(@Nonnull WorkflowRun run,
                         return (lastNode.getError() == null) ? GenericStatus.SUCCESS : GenericStatus.FAILURE;
                     }
                 }
+
+                if (lastNode instanceof StepStartNode) {
+                    ExecutorTaskInfoAction execAction = lastNode.getAction(ExecutorTaskInfoAction.class);
+                    if (execAction != null) {
+                        if (execAction.isQueued()) {
+                            return GenericStatus.QUEUED;
+                        } else if (execAction.isCancelled()) {
+                            return GenericStatus.ABORTED;
+                        } else {
+                            return GenericStatus.IN_PROGRESS;
+                        }
+                    }
+                }
+
                 PauseAction pauseAction = lastNode.getAction(PauseAction.class);
-                return (isPendingInput(run) &&
-                        pauseAction != null && pauseAction.getCause().equals("Input"))
-                        ? GenericStatus.PAUSED_PENDING_INPUT : GenericStatus.IN_PROGRESS;
+                if (isPendingInput(run) &&
```

Comment:
You'll take my ternary operators from my cold, dead hands, mate.  COLD.  DEAD.  HANDS. :skull:    

(Kidding, I think this probably makes this block more readable for others in this case).